tests/FSharp.Compiler.ComponentTests/EmittedIL/RealInternalSignature/ClassTypeInitialization.fs

```diff
@@ -842,6 +847,325 @@ namespace Microsoft.FSharp.Core.CompilerServices
 """
         |> asExe
         |> withRealInternalSignature realSig
-        |> withOptimize
+        |> withOptimization
+        |> compileAndRun
+        |> shouldSucceed
+
+    [<InlineData(true, true)>]          // RealSig Optimize
+    [<InlineData(true, false)>]         // RealSig NoOptimize
+    [<InlineData(false, true)>]         // Regular Optimize
+    [<InlineData(false, false)>]        // Regular NoOptimize
+    [<Theory>]
+    let ``Generic class with closure with constraints`` (realSig, optimize) =
+        let withOptimization compilation =
+            if optimize then compilation |> withOptimize
+            else compilation |> withNoOptimize
+
+        FSharp """
+namespace Test
+open System
+
+module RuntimeHelpers =
+    [<Sealed>]
+    type MyType<'A,'B when 'B :> seq<'A>>(sources: seq<'B>) =
+
+        member x.MoveNext() =
+            let rec takeInner c =
+                let rec takeOuter b =
+                    if b.ToString () = "1" then failwith "Oops"
+                    if sources |> Seq.length > 10 then
+                        sources |> Seq.skip 10
+                    else
+                        sources
+                if c.ToString() = "1" then failwith "Oops"
+                if sources |> Seq.length < 5 then
+                    sources
+                else
+                    takeOuter 7
+            takeInner 3
+
+open RuntimeHelpers
+module doIt =
+    let x = seq {  ArraySegment([|1uy; 2uy; 3uy|]); ArraySegment([|1uy; 2uy; 3uy|]); ArraySegment([|1uy; 2uy; 3uy|]); }
+    let enumerator = x |> MyType<_,_>
+    for i in enumerator.MoveNext() do
+        printfn "%A" i
+    """
+        |> asExe
+        |> withRealInternalSignature realSig
+        |> withOptimization
         |> compileAndRun
         |> shouldSucceed
+
+    [<InlineData(true, true)>]          // RealSig Optimize
+    [<InlineData(true, false)>]         // RealSig NoOptimize
+    [<InlineData(false, true)>]         // Regular Optimize
+    [<InlineData(false, false)>]        // Regular NoOptimize
+    [<Theory>]
+    let ``AgedLookup`` (realSig, optimize) =
+        let withOptimization compilation =
+            if optimize then compilation |> withOptimize
+            else compilation |> withNoOptimize
+
+        FSharp """
+namespace Internal.Utilities.Collections
```

Comment:
This code snippet is in my opinion big enough to justify living in its own file instead of being in an inline string.

Reply:
> This code snippet is in my opinion big enough to justify living in its own file instead of being in an inline string.

I'm not actually a fan of those files, but I can extract it.